### PR TITLE
Fix gpt example attention mask

### PR DESCRIPTION
**Pull request by gongel**
### PR types

Bug fixes
### PR changes

Examples
### Description

- Update example in sync with #3133

### CODE REVIEW

**Review by guoshengCS — DISMISSED** [review]
LGTM. 这个还请 @ZHUI 再确认下呢，transformers/gpt/modeling.py 做了不兼容的修改 https://github.com/PaddlePaddle/PaddleNLP/pull/3133 ，当前支持0、1的输入mask，这里之前的使用漏掉了

File: model_zoo/gpt/dataset.py [hunk 1]
```diff
@@ -442,17 +442,13 @@ def _construct_sample(self, tokens):
...
-            (1, seq_length, seq_length))
 
         # The pad and eos tokens do not contribute the loss
         loss_mask = np.ones(seq_length, dtype="float32")
         loss_mask[np.where(np.array(tokens) == self.eos_id)] = 0.0
         position_ids = np.arange(0, seq_length, dtype="int64")
 
-        attention_mask = (attention_mask - 1.0) * 1e9
-        attention_mask = attention_mask.astype("float32")
+        attention_mask = loss_mask
```
Comment (ZHUI) [inline model_zoo/gpt/dataset.py]: 还是有点不妥，`attention_mask = loss_mask` 屏蔽掉了 `eos`的token，但是很多下游，是用`eos`表示一些token的生成和结束。

Comment (gongel) [reply model_zoo/gpt/dataset.py]: 经沟通，该example没有做pad，所以attention mask全1即可。

**Review by ZHUI — COMMENTED** [review]
辛苦这里也改一下吧
 https://github.com/PaddlePaddle/PaddleNLP/blob/develop/model_zoo/gpt/run_pretrain_static.py#L56

**Comment by gongel** [comment]
> 辛苦这里也改一下吧 https://github.com/PaddlePaddle/PaddleNLP/blob/develop/model_zoo/gpt/run_pretrain_static.py#L56

DONE, thx

**Review by ZHUI — APPROVED** [review]
LGTM
